sdk/storage/test/datalake/file_system_client_test.cpp

```diff
@@ -0,0 +1,238 @@
+  std::string FileSystemClientTest::m_DirectoryA;
+  std::string FileSystemClientTest::m_DirectoryB;
+
+  void FileSystemClientTest::SetUpTestSuite()
+  {
+    m_fileSystemName = LowercaseRandomString(10);
+    m_fileSystemClient = std::make_shared<DataLake::FileSystemClient>(
+        DataLake::FileSystemClient::CreateFromConnectionString(
+            ADLSGen2ConnectionString(), m_fileSystemName));
+    m_fileSystemClient->Create();
+
+    m_DirectoryA = LowercaseRandomString(10);
+    m_DirectoryB = LowercaseRandomString(10);
+    for (unsigned i = 0; i < c_PATH_TEST_SIZE; ++i)
+    {
+      {
+        auto name = m_DirectoryA + LowercaseRandomString(10);
+        m_fileSystemClient->GetPathClient(m_DirectoryA + "/" + name).CreateAsFile();
+        m_PathNameSetA.emplace_back(std::move(name));
+      }
+      {
+        auto name = m_DirectoryB + LowercaseRandomString(10);
+        m_fileSystemClient->GetPathClient(m_DirectoryB + "/" + name).CreateAsFile();
+        m_PathNameSetB.emplace_back(std::move(name));
+      }
+    }
+  }
+
+  void FileSystemClientTest::TearDownTestSuite() { m_fileSystemClient->Delete(); }
+
+  std::vector<DataLake::Path> FileSystemClientTest::ListAllPaths(
+      bool recursive,
+      const std::string& directory)
+  {
+    std::vector<DataLake::Path> result;
+    std::string continuation;
+    DataLake::ListPathsOptions options;
+    if (!directory.empty())
+    {
+      options.Directory = directory;
+    }
+    do
+    {
+      auto response = m_fileSystemClient->ListPaths(recursive, options);
+      result.insert(result.end(), response.Paths.begin(), response.Paths.end());
+      if (response.Continuation.HasValue())
+      {
+        continuation = response.Continuation.GetValue();
+        options.Continuation = continuation;
+      }
+    } while (!continuation.empty());
+    return result;
+  }
+
+  DataLake::DataLakeHttpHeaders FileSystemClientTest::GetInterestingHttpHeaders()
+  {
+    static DataLake::DataLakeHttpHeaders result{
+        std::string("no-cache"),
+        std::string("attachment"),
+        std::string("deflate"),
+        std::string("en-US"),
+        std::string("text/html; charset=UTF-8")};
+    return result;
+  }
+
+  TEST_F(FileSystemClientTest, CreateDeleteFileSystems)
+  {
+    {
+      // Normal create/delete.
+      std::vector<DataLake::FileSystemClient> fileSystemClient;
+      for (int32_t i = 0; i < 5; ++i)
+      {
+        auto client = DataLake::FileSystemClient::CreateFromConnectionString(
+            ADLSGen2ConnectionString(), LowercaseRandomString(10));
+        EXPECT_NO_THROW(client.Create());
+        fileSystemClient.emplace_back(std::move(client));
+      }
+      for (const auto& client : fileSystemClient)
+      {
+        EXPECT_NO_THROW(client.Delete());
+      }
+    }
+    {
+      // Normal delete with access condition.
+      std::vector<DataLake::FileSystemClient> fileSystemClient;
+      for (int32_t i = 0; i < 5; ++i)
```

Comment:
We never say `int` is forbidden everywhere. This `i` veriable is not gonna be sent out over the wire.

Reply:
True but I don't think int32_t is forbidden.

Reply:
whatever